include/boost/geometry/formulas/distance_point_segment.hpp

```diff
@@ -0,0 +1,345 @@
+// Boost.Geometry
+
+// Copyright (c) 2016-2017 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_CROSS_TRACK_GEO_HPP
+#define BOOST_GEOMETRY_FORMULAS_CROSS_TRACK_GEO_HPP
+
+#include <boost/geometry/formulas/result_direct.hpp>
+
+#ifndef BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS
+#define BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS 100
+#endif
+
+/*!
+\brief Algorithm to compute the distance between a segment and a point using
+       direct and inverse geodesic problems as subroutines. The algorithm
+       approximates the distance by an iterative Newton method.
+\see C.F.F.Karney - Geodesics on an ellipsoid of revolution,
+      https://arxiv.org/abs/1102.1215
+*/
+
+namespace boost { namespace geometry { namespace formula
+{
+
+template
+<
+        typename CT,
+        typename Units,
+        template <typename, bool, bool, bool, bool ,bool> class Inverse,
+        template <typename, bool, bool, bool, bool> class Direct,
+        bool EnableClosestPoint = false
+>
+class distance_point_segment{
+
+public:
+
+    typedef Inverse<CT, true, false, false, true, true> inverse_distance_type;
+    typedef Inverse<CT, false, true, false, false, false> inverse_azimuth_type;
+    typedef Direct<CT, true, false, false, false> direct_distance_type;
+
+    struct result_distance_point_segment
+    {
+        result_distance_point_segment()
+            : distance(0)
+            , closest_point_lon(0)
+            , closest_point_lat(0)
+        {}
+
+        CT distance;
+        CT closest_point_lon;
+        CT closest_point_lat;
+    };
+
+    template <typename Spheroid>
+    result_distance_point_segment
+    static inline non_iterative_case(CT lon1, CT lat1, //p1
+                                     CT lon2, CT lat2, //p2
+                                     Spheroid const& spheroid)
+    {
+        result_distance_point_segment result;
+        result.distance = inverse_distance_type::apply(lon1, lat1,
+                                                       lon2, lat2,
+                                                       spheroid).distance;
+        if (EnableClosestPoint)
+        {
+            result.closest_point_lon = lon1;
+            result.closest_point_lat = lat1;
+        }
+        return result;
+    }
+
+    template <typename Spheroid>
+    result_distance_point_segment
+    static inline apply(CT lon1, CT lat1, //p1
+                        CT lon2, CT lat2, //p2
+                        CT lon3, CT lat3, //query point p3
+                        Spheroid const& spheroid)
+    {
+        CT earth_radius = (CT(2) * spheroid.get_radius<1>()
+                           + spheroid.get_radius<2>()) / CT(3);
```

Comment:
I just added `mean_radius()` formula to develop which could be used here, also the variable could be const. But this could be changed later after the PR is merged.

Reply:
I used the mean_radius() formula here, thanks. A side remark: is there a reason to call the argument of mean_radius() "geometry"; as far as I understand it is a coordinate system reference model, i.e. a sphere or spheroid?

Reply:
It's just a general name for parameter of multiple types. It could've been called `Model` as well. But a geometry could also be passed into this function, like, e.g. nsphere from extensions or some n-dimensional ellipsoid in the future, though very unlikely we'll have that.